### PR TITLE
mirror: T3169: Add smoke testing and fix dependency issues

### PR DESCRIPTION
**Pull request by jack9603301**
## Change Summary

Add smoke testing and fix dependency issues

## Types of changes

 
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

https://phabricator.vyos.net/T3169

## Component(s) name

mirror

## Proposed changes

Add smoke testing and fix dependency issues

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by jack9603301** [comment]
Smoke test passed in local environment

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 1]
```diff
@@ -51,14 +51,16 @@ def setUp(self):
...
+            self._mirror_interfaces.append('bond1')
 
         self._options['br0'] = []
         for member in self._members:
             self._options['br0'].append(f'member interface {member}')
 
     def tearDown(self):
-        self.session.delete(['interfaces','bonding'])
-        self.session.delete(['interfaces','dummy'])
+        self.session.delete(['interfaces','bonding','bond1'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: Why is this change required?

File: smoketest/scripts/cli/test_interfaces_ethernet.py [hunk 2]
```diff
@@ -66,6 +66,10 @@ def setUp(self):
                 mac = read_file(f'/sys/class/net/{interface}/address')
             self._macs[interface] = mac
 
+        # Creating test interfaces for port mirroring
+        self.session.set(['interfaces','dummy','dum100'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_ethernet.py]: Please use spaces for list items.

File: smoketest/scripts/cli/test_interfaces_bonding.py [hunk 3]
```diff
@@ -50,6 +50,14 @@ def setUp(self):
         for member in self._members:
             self._options['bond0'].append(f'member interface {member}')
 
+        # Creating test interfaces for port mirroring
+        self.session.set(['interfaces','dummy','dum10'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bonding.py]: Use spaces for list items.

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 1 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py:61]: In fact, it doesn't need to be modified here. However, in order to ensure that the redundant interface will not be deleted, it is improved by the way. Do you want to restore the original deletion instruction?

Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: In fact, there is no need to change it here. But in order to ensure that the redundant interface is not deleted, it is improved

File: smoketest/scripts/cli/test_interfaces_bonding.py [hunk 4]
```diff
@@ -50,6 +50,14 @@ def setUp(self):
         for member in self._members:
             self._options['bond0'].append(f'member interface {member}')
 
+        # Creating test interfaces for port mirroring
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bonding.py:54]: Speciy `dum10` as value and probably use a dum interface number > 10000

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 5]
```diff
@@ -51,6 +51,7 @@ def setUp(self):
             self.session.set(['interfaces','bonding','bond1','member','interface','dum0'])
             self.session.set(['interfaces','bonding','bond1','member','interface','dum1'])
             self._members.append('bond1')
+            self._mirror_interfaces.append('bond1')
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: This should not be done for ALL testscases, rather limit this to the specific test.

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_interfaces_bonding.py [hunk 6]
```diff
@@ -50,6 +51,15 @@ def setUp(self):
...
+        for mon_intf in self._mirror_interfaces:
+            if 'dum' in mon_intf:
+                self.session.set(['interfaces', 'dummy', mon_intf])
+    
+    def tearDown(self):
+        self.session.delete(['interfaces','dummy','dum10'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bonding.py]: Why `dum10`, you really need to be more careful!

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 7]
```diff
@@ -35,6 +35,7 @@ def setUp(self):
 
         self._base_path = ['interfaces', 'bridge']
         self._interfaces = ['br0']
+        self._mirror_interfaces = ['bond1']
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: Who creates bond1?

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_interfaces_bonding.py (hunk 6 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bonding.py]: Oh, I haven't tested it yet. I'm just going to 'push' it first, and I'm preparing for the test

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 7 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: The code for creating `bond1` is below. I share `bond1`

**Comment by jack9603301** [comment]
The interface number is now greater than 10000

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 8]
```diff
@@ -37,8 +37,10 @@ def setUp(self):
 
         self._base_path = ['interfaces', 'bridge']
         self._interfaces = ['br0']
+
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py:40]: Why do we need a blank line?

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 9]
```diff
@@ -48,9 +50,22 @@ def setUp(self):
                 if not '.' in tmp:
                     self._members.append(tmp)
 
+            self.session.set(['interfaces','dummy','dum10001'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py:53]: You continously do not follow the code path. I requested to move this out of the general interface setup for multiple times now.
